superset/connectors/sqla/models.py

```diff
@@ -90,6 +90,7 @@ class AnnotationDatasource(BaseDatasource):
 
     cache_timeout = 0
     changed_on = None
+    type = "annotation"
```

Comment:
@villebro I needed to add this to after making the `BaseDatasource.type` an abstract property otherwise [this](https://github.com/apache/incubator-superset/blob/master/superset/connectors/base/models.py#L111) would barf. Note even though unused I gather having `<id>__annotation` is more accurate that `<id>__None` which it was previously.

cc: @etr2460